src/plugins/dashboard/public/application/dashboard_app_controller.tsx

```diff
@@ -328,6 +327,22 @@ export class DashboardAppController {
       dashboardStateManager.getPanels().forEach((panel: SavedDashboardPanel) => {
         embeddablesMap[panel.panelIndex] = convertSavedDashboardPanelToPanelState(panel);
       });
+
+      // If the incoming embeddable state's id already exists in the embeddables map, replace the input, retaining the existing gridData for that panel.
+      if (incomingEmbeddable?.embeddableId && embeddablesMap[incomingEmbeddable.embeddableId]) {
+        const originalPanelState = { ...embeddablesMap[incomingEmbeddable.embeddableId] };
```

Comment:
Maybe this copy can be skipped? The line next assigns a new object to it.
An assignment should be enough, I think.

Reply:
Good call - removed this extra copy in the next version: https://github.com/elastic/kibana/pull/77561